src/js/server/initblock.js

```diff
@@ -0,0 +1,130 @@
+// # What is this?
+//
+// The `initBlock` Handlebars helper constructs a client side component given a list of grades, including all
+// wiring required to ensure that:
+//
+//   1. Only one renderer is created and used in all child components
+//   2. Child components are only created once templates have been loaded.
+//
+// The end result is a nested structure, starting with a `pageComponent` variable that has a global renderer.  This
+// component has a `requireRenderer` child component, which is created only when the templates are available, and holds
+// a `pageComponent` component constructed from the grades passed to the helper (see below).
+//
```

Comment:
Should add a reference here to the gpii.templates.templateManager component since presumably this infrastructure can't be used without it